bite-size-notes/week-5-d-various-mpc.md

```diff
@@ -0,0 +1,285 @@
+
+The challenge of Robust MPC is that its performance deteriorates in environments with high uncertainty. 
+For example, if there is significant noise in the sensors, it is necessary to expand the range of the reachability set assuming that the prediction errors will be large. 
+Therefore, in order to ensure that the entire set satisfies the constraints, extremely conservative control is required, leading to a decrease in the robot's performance in executing its intended task.
+
+## Stochastic MPC
+Consider a system given by:
+
+$$
+x_{k+1} = Ax_k + Bu_k + w_k\\
+u_k = \phi_k(X_k),\ X_k=[x_0,\ldots,x_k]\\
+x_k\in R^n, u_k\in R^m, w_k \in R^n
+$$
+
+where $T$ is finite horizon and $x_0, w_0, \ldots, w_{T-1}$ are random variables.
+Then, the objective function is defined as follows:
+
+$$
+J = \mathbb{E}(\sum_{k=0}^{T-1} l_k(x_k, u_k)+l_T(x_T))
+$$
+
+where $\mathbb{E}$ is expectation function, and $l_k$ and $l_T$ are convex.
+The objective function $J$ depends on the control policy $\phi_k$, so the goal of this problem is to choose $\phi_k$ that minimizes $J$.
+Since the variable is a function $\phi_k$, this problem is an infinite-dimensional problem.
+
+### Linear Quadratic Stochastic MPC
+Stochastic MPC can analytically find solutions in special cases. We make the following assumptions:
+
+$$
+\begin{aligned}
+&x_0, w_0, \ldots, w_{T-1}\ \mathrm{are\ independent}\\
+&\mathbb{E}(x_0) = 0,\ \mathbb{E}(w_k) = 0,\ \mathbb{E}(x_0x_0^T) = \Sigma,\ \mathbb{E}(w_kw_k^T) = W_k 
+\end{aligned}
+$$
+
+And suppose the objective function is as follows:
+
+$$
+l_k(x_k, u_k) = x_k^TQ_kx_k + u_k^TR_ku_k,\ Q_k\succeq 0,\ R_k\succ 0\\
+l_T(x_T) = x_T^TQ_kx_T,\ Q_T\succeq 0
+$$
+
+Assuming the cost function is quadratic as follows:
+
+$$
+V_k(x_k) = x_k^TP_kx_k + q_k,\ k=0,\ldots,T
+$$
+
+The value function at time step $k$ can be expressed as follows:
+
+$$
+V_k(x_k) = \inf_{u_k} x_k^TQ_kx_k + u_k^TR_ku_k + \mathbb{E}((Ax_k + Bu_k + w_k)^TP_{k+1}(Ax_k + Bu_k + w_k) + q_{k+1})
+$$
+
+This results in the following:
+
+$$
+P_k = A^TP_{k+1}A-A^TP_{k+1}B(B^TP_{k+1}B+R_k)^{-1}B^TP_{k+1}A+Q_k\\
+q_k = q_{k+1} + \mathrm{Tr}(W_kP_{k+1})
+$$
+
+Therefore, by setting $P_T = Q_T$ and iteratively calculating $P_{T-1}, \ldots, P_0$, the optimal policy can be computed as follows, independent of the noise term.
+
+$$
+\phi_k^*(x_k) = K_kx_k\\
+K_k = -(B^TP_{k+1}B+R_k)^{-1}B^TP_{k+1}A
+$$
+
+## Conclusion
+Model Predictive Control (MPC) and its variants offer robust and flexible control strategies for autonomous systems. 
+Linear MPC is effective for linear dynamics, while Nonlinear MPC directly optimizes nonlinear models. 
+Robust MPC handles disturbances by considering reachable states, and Stochastic MPC optimizes control under probabilistic uncertainties. 
+Understanding these variants is key for selecting the most suitable strategy for different applications.
```

Comment:
Great preview of most used MPC variants, maybe  to conclude you can say there are other MPC variants are available.